art/classifiers/detector_classifier.py

```diff
@@ -38,23 +38,30 @@ class DetectorClassifier(ClassifierNeuralNetwork, ClassifierGradients, Classifie
     More details in https://arxiv.org/abs/1705.07263
     """
 
-    def __init__(self, classifier, detector, defences=None, preprocessing=(0, 1)):
+    def __init__(self, classifier, detector, preprocessing_defences=None, postprocessing_defences=None,
+                 preprocessing=(0, 1)):
         """
         Initialization for the DetectorClassifier.
 
         :param classifier: A trained classifier.
         :type classifier: :class:`.Classifier`
         :param detector: A trained detector applied for the binary classification.
         :type detector: :class:`.Detector`
-        :param defences: Defences to be activated with the classifier.
-        :type defences: `str` or `list(str)`
+        :param preprocessing_defences: Preprocessing defence(s) to be activated with the classifier. Not applicable
+               in this classifier.
+        :type preprocessing_defences: :class:`.Preprocessor` or `list(Preprocessor)` instances
+        :param postprocessing_defences: Postprocessing defence(s) to be activated with the classifier.
+        :type postprocessing_defences: :class:`.Postprocessor` or `list(Postprocessor)` instances
         :param preprocessing: Tuple of the form `(subtractor, divider)` of floats or `np.ndarray` of values to be
                used for data preprocessing. The first value will be subtracted from the input. The input will then
-               be divided by the second one.
+               be divided by the second one. Not applicable in this classifier.
```

Comment:
Should we raise an exception if preprocessing is not None?

Reply:
Solved.